src/loaders/genome_collection/parse_tool_results.py

```diff
@@ -305,6 +308,38 @@ def _create_import_files(root_dir: str, file_name: str, docs: list[dict[str, Any
         convert_to_json(docs, f)
 
 
+def _process_mash_tool(root_dir: str,
+                       kbase_collection: str,
+                       load_ver: str):
+    # merge and create a single sketch file from result sketch files generated by mash sketch
+
+    result_dir = _locate_dir(root_dir, kbase_collection, load_ver, tool='mash')
+    batch_dirs = _get_batch_dirs(result_dir)
+
+    sketch_files = list()
+    for batch_dir in batch_dirs:
+        data_ids = [item for item in os.listdir(os.path.join(result_dir, batch_dir)) if
+                    os.path.isdir(os.path.join(result_dir, batch_dir, item))]
+        for data_id in data_ids:
+            data_dir = Path(result_dir, batch_dir, data_id)
+            with open(data_dir / loader_common_names.MASH_METADATA, 'r') as file:
+                metadata = json.load(file)
+            # Append '.msh' to the source file name to generate the sketch file name (default by Mash sketch)
+            sketch_file = metadata['source_file'] + '.msh'
+            if not os.path.exists(sketch_file):
+                raise ValueError(f'Unable to locate the sketch file: {sketch_file} for genome: {data_id}')
+            sketch_files.append(sketch_file)
+
+    import_dir = Path(root_dir, IMPORT_DIR)
+    os.makedirs(import_dir, exist_ok=True)
+    mash_output_prefix = import_dir / f'{kbase_collection}_{load_ver}_merged_sketch'
+
+    # run mash paste
+    command = ['mash', 'paste', mash_output_prefix] + sketch_files
```

Comment:
If there are 300k input files is this going to work? Does mash paste take a list of files like sketch does?

Reply:
👍  switch to use `-l file`.

Reply:
:+1: